packages/server/src/api/controllers/public/rows.ts

```diff
@@ -0,0 +1,75 @@
+import { default as rowController } from "../row"
+import { addRev } from "./utils"
+import { Row } from "../../../definitions/common"
+
+// makes sure that the user doesn't need to pass in the type, tableId or _id params for
+// the call to be correct
+function fixRow(row: Row, params: any) {
+  if (!params || !row) {
+    return row
+  }
+  if (params.rowId) {
+    row._id = params.rowId
+  }
+  if (params.tableId) {
+    row.tableId = params.tableId
+  }
+  if (!row.type) {
+    row.type = "row"
+  }
+  return row
+}
+
+export async function search(ctx: any) {
+  let { sort, paginate, bookmark, limit, query } = ctx.request.body
+  // update the body to the correct format of the internal search
+  if (!sort) {
+    sort = {}
+  }
+  ctx.request.body = {
+    sort: sort.column,
+    sortType: sort.type,
+    sortOrder: sort.order,
+    bookmark,
+    paginate,
+    limit,
+    query,
+  }
+  await rowController.search(ctx)
+}
+
+export async function create(ctx: any) {
+  ctx.request.body = fixRow(ctx.request.body, ctx.params)
+  await rowController.save(ctx)
+  ctx.body = { row: ctx.body }
```

Comment:
I really like the idea of wrapping the raw response in an object. I think there's room here to generify the wrapping by using a `data` key instead of `row`, `table` etc, as we already know the resource type by the path and this will make it easier for clients to consume. 

e.g. stripe api 
```js
{
  "object": "list",
  "url": "/v1/products",
  "has_more": false,
  "data": [
    {
      "id": "prod_LEuBjywZYxGfuH",
      "object": "product",
      "active": true,
      "created": 1646120649,
      "description": null
   }]
}
```

Usually you only see this type on object wrapping on collections endpoints btw, for single object responses it's fine to just return the resource directly

Reply:
Personally I'm not a fan of using a generic wrapper, I also really like wrapping in general, theres no negative to the user experience and it allows us to add on anything extra in the future, I've had too many issues with single object responses returning directly in the past.